cms/djangoapps/contentstore/views/course.py

```diff
@@ -625,72 +622,13 @@ def course_index(request, course_key):
-            if course_block.start != DEFAULT_START_DATE
-            else _("Set Date")
-        )
-
-        settings_url = reverse_course_url('settings_handler', course_key)
-
-        try:
-            current_action = CourseRerunState.objects.find_first(course_key=course_key, should_display=True)
-        except (ItemNotFoundError, CourseActionStateItemNotFoundError):
-            current_action = None
-
-        deprecated_block_names = [block.name for block in deprecated_xblocks()]
-        deprecated_blocks_info = _deprecated_blocks_info(course_block, deprecated_block_names)
+    if not course_block:
+        raise Http404
+    if use_new_course_outline_page(course_key):
+        return redirect(get_course_outline_url(course_key))
 
-        frontend_app_publisher_url = configuration_helpers.get_value_for_org(
-            course_block.location.org,
-            'FRONTEND_APP_PUBLISHER_URL',
-            settings.FEATURES.get('FRONTEND_APP_PUBLISHER_URL', False)
-        )
-        # gather any errors in the currently stored proctoring settings.
-        advanced_dict = CourseMetadata.fetch(course_block)
-        proctoring_errors = CourseMetadata.validate_proctoring_settings(course_block, advanced_dict, request.user)
-
-        user_clipboard = content_staging_api.get_user_clipboard_json(request.user.id, request)
-
-        return render_to_response('course_outline.html', {
-            'language_code': request.LANGUAGE_CODE,
-            'context_course': course_block,
-            'lms_link': lms_link,
-            'sections': sections,
-            'course_structure': course_structure,
-            'initial_state': course_outline_initial_state(locator_to_show, course_structure) if locator_to_show else None,  # lint-amnesty, pylint: disable=line-too-long
-            'initial_user_clipboard': user_clipboard,
-            'rerun_notification_id': current_action.id if current_action else None,
-            'course_release_date': course_release_date,
-            'settings_url': settings_url,
-            'reindex_link': reindex_link,
-            'deprecated_blocks_info': deprecated_blocks_info,
-            'notification_dismiss_url': reverse_course_url(
-                'course_notifications_handler',
-                current_action.course_key,
-                kwargs={
-                    'action_state_id': current_action.id,
-                },
-            ) if current_action else None,
-            'frontend_app_publisher_url': frontend_app_publisher_url,
-            'mfe_proctored_exam_settings_url': get_proctored_exam_settings_url(course_block.id),
-            'advance_settings_url': reverse_course_url('advanced_settings_handler', course_block.id),
-            'proctoring_errors': proctoring_errors,
-            'taxonomy_tags_widget_url': get_taxonomy_tags_widget_url(course_block.id),
-        })
+    course_index_context = get_course_index_context(request, course_key, course_block)
```

Comment:
@navinkarkera I believe the reason we had to revert this PR is because this line moved out from being indented in the with modulestore().bulk_operations(course_key): resulting in 1000s of individual operations for courses.

Reply:
I'm not exactly sure why (because you did put your call to get the course in its own bulk operation, but 🤷 )

Reply:
@connorhaugh That is weird. Like you pointed out, we do have the get_course call under its own bulk operation. It might be due to `sections = course_block.get_children()` at a later point in `get_course_index_context` function.  🤷

I created a new PR:https://github.com/openedx/edx-platform/pull/33943 to add this API and fix this issue. I am not sure how to test whether this issue is resolved by moving the whole function call under bulk_operations.

Reply:
@navinkarkera thanks for following up on this promptly. I'll leave a comment in the new PR.